src/codegeneration/TemplateLiteralTransformer.js

```diff
@@ -38,74 +37,79 @@ import {
   createArrayLiteralExpression,
   createBinaryExpression,
   createCallExpression,
-  createIdentifierExpression,
+  createMemberExpression,
   createOperatorToken,
-  createParenExpression,
-  createStringLiteral
+  createParenExpression
 } from './ParseTreeFactory.js';
-import {parseExpression} from './PlaceholderParser.js';
+
+function createStringLiteralExpression(loc, str) {
+  return new LiteralExpression(loc, new LiteralToken(STRING, str, loc));
+}
 
 /**
- * @param {ParseTree} tree
+ * Generetes the runtime call to create the template object.
+ * @param {Array<ParseTree>} elements
  * @return {ParseTree}
  */
-function createCallSiteIdObject(tree) {
-  let elements = tree.elements;
-  let cooked = createCookedStringArray(elements);
-  let raw = createRawStringArray(elements);
-  return parseExpression `Object.freeze(Object.defineProperties(${cooked}, {
-    raw: {
-      value: Object.freeze(${raw})
+function createGetTemplateObject(elements) {
+  // In case cooked and raw have the same value (which is quite common) we
+  // do not need to repeat the array literal.
+  //
+  // $traceurRuntime.getTemplateObject(['a', 'b'])
+  // $traceurRuntime.getTemplateObject(['\\n'], ['\n'])
+
```

Comment:
first example has a single array literal, because the cooked values and raw values are identical. The second example has 2 array literals, because the cooked string is different from the raw string

(at least, I think that's what this is getting at).